### ISSUE TITLE
Sailfish OS: Huge amount of map tile image files slows down MTP connection of the device

### ISSUE THREAD

**Issue by wanderer1988** [M4rtinK/modrana#99]
I recently investigated on my jolla that storing map tiles as files can cause troubles with the MTP connection. I reported this issue here: https://together.jolla.com/question/102084/huge-amount-of-image-files-ie-cached-map-tiles-slows-down-or-blocks-mtp-connection/. In my case I had over 83000 files in ~/.local/harbour-modrana/maps which had negative influence on the MTP connection.

Of course this is rather a bug of Sailfish OS than modrana, but it could make sense to store map tiles in SQLite databases instead of files by default until this issue is resolved by jolla. Also a possibility to delete map tiles in the app could be useful.

**Comment by M4rtinK** [comment]
This really looks like a quite bad OS-side issue.

As for switching to using Sqlite tile storage by default - modRana currently only supports loading tiles either from files or Sqlite, so if I changed this setting for existing installations it would mean that the already stored tiles would become unavailable to users while still occupying storage & causing the mentioned MPT issue. ModRana has also some support for using a platform-default (this is used for example from the beginning on Android for similar reasons - brain-dead image indexers) for tile storage, but that would only influence new installations and would not help current modRana users.

In short - I don't see any simple short term modRana-side solution fixing this issue for both current and future modRana users. On the other hand fixing this gross platform-level issue will fix it for all users of modRana and other mapping and image handling apps.

**Comment by M4rtinK** [comment]
Not to end on just a negative note, I have some mid-term plans for improved tile handling in modRana unrelated to this issue:
* moving tile handling to a separate Python module
 * this should make it easier to test it with unit tests
 * it would also make it possible to use the module outside of modRana for stuff like tile manipulation scripts and tile repositories
* making it possible to load tiles for a single layer from multiple sources (say from files and a sqlite database, from multiple folders (internal storage, SDcard, etc.), etc.)
* moving tiles between file storage and sqlite databases (and possibly other tile storage modes)
* tile storage accounting and manipulation support in the modRana GUI
 * showing space occupied by tiles per layer
   * this could take quite a long time for file storage (due to thousands of files and folders) and is instant for Sqlite tile storage
 * deleting tiles per layer
 * conversions
And as always - **patches welcome**! - as this is hardly the only thing I would like to add to modRana. :)
